spec.md

```diff
@@ -23,42 +23,25 @@ A standard token interface is a basic building block for many applications on th
    	totalSupply : Nat; // token total supply
    	owner : Principal; // token owner
    	fee : Nat; // fee for update calls
-   	feeTo : Principal; // fee receiver
    }
    ```
 
-2. Status: token status info
-
-   ```js
-   type Status = {
-   	historySize : Nat; // total number of history transactions
-   	deployTime: Time.Time; // token canister deploy time in nanoseconds
-   	holderNumber : Nat; // token holder number
-   	cycles : Nat; // token canister cycles balance
-   }
-   ```
-
-3. TxReceipt: receipt for update calls, contains the transaction index or an error message
+2. TxReceipt: receipt for update calls, contains the transaction index or an error message
 
    ```js
    type TxReceipt = Result.Result<Nat, {
    	#InsufficientBalance;
    	#InsufficientAllowance;
    }>;
-   ```
-
-4. TxRecord: history transaction record
+   
+3. TxRecord: history transaction record
 
    ```js
    public type Operation = {
-   	#mint;
-   	#burn;
    	#transfer;
    	#approve;
-   	#init;
    };
    public type TxRecord = {
-   	caller: Principal; // caller of the transaction
```

Comment:
if we don't require mint/burn, I think we don't need `caller` here:
for transferFrom, transfer, approve methods `from` has to be the same as `caller`, so caller is not required

Reply:
The caller for transferFrom is the canister, not the `from`

Reply:
I can approve you to take 100 tokens from my account by calling the `approve` function. 
`approve(Daniel, CCyan, 100)`
Caniester will check if `from` = caller (and also check many things)

after which you can call
`transferFrom(Daniel, CCyan, 50)`
canister will check if `to` = caller (and also check many things)

So you are right that the canister doesn't check `from` = caller for transferFrom

However we really only need to store 2 principals(not 3 - from/to/originator) for all 3 calls. So maybe we could just give the appropriate names to those 2 principal(not

Reply:
updated above